src/pip/_internal/index.py

```diff
@@ -871,23 +965,6 @@ def _find_name_version_sep(egg_info, canonical_name):
     raise ValueError("{} does not match {}".format(egg_info, canonical_name))
 
 
-def _egg_info_matches(egg_info, canonical_name):
-    """Pull the version part out of a string.
-
-    :param egg_info: The string to parse. E.g. foo-2.1
-    :param canonical_name: The canonicalized name of the package this
-        belongs to.
-    """
-    try:
-        version_start = _find_name_version_sep(egg_info, canonical_name) + 1
-    except ValueError:
-        return None
-    version = egg_info[version_start:]
-    if not version:
-        return None
-    return version
-
-
```

Comment:
I would do moves of functions in a separate PR since it makes it easier to review the patches with changes. I'm happy to accept those quickly if you give me a heads up.